src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -277,7 +321,38 @@ protected async Task<(Guid, string projectName, SourceText)> GetGuidAndProjectNa
 
                 return new DocumentSpanEntry(
                     this, definitionBucket, documentSpan, spanKind,
-                    projectName, guid, sourceText, classifiedSpansAndHighlightSpan);
+                    projectName, guid, sourceText, classifiedSpansAndHighlightSpan, GetAggregatedCustomColumnsData());
+
+                // Local functions.
+                ImmutableDictionary<string, string> GetAggregatedCustomColumnsData()
+                {
+                    // Aggregate multidictionary values to get column display values. For example, below input:
+                    //
+                    // {
+                    //   { "Column1", {"Value1", "Value2"} },
+                    //   { "Column2", {"Value3", "Value4"} }
+                    // }
+                    //
+                    // will transform to:
+                    //
+                    // {
+                    //   { "Column1", "Value1, Value2" },
+                    //   { "Column2", "Value3, Value4" }
+                    // }
+
+                    if (customColumnsDataOpt == null || customColumnsDataOpt.Count == 0)
+                    {
+                        return ImmutableDictionary<string, string>.Empty;
+                    }
+
+                    return customColumnsDataOpt.ToImmutableDictionary(
+                        keySelector: kvp => kvp.Key,
+                        elementSelector: kvp => GetCustomColumn(kvp.Key).GetDisplayStringForColumnValues(kvp.Value));
+
```

Comment:
1. extra newline.
2. is this info cached?  we can easily have thousands of results for some searches.  would prefer to be a good allocation citizen.

Reply:
`GetCustomColumn` uses the name to column cache in the editor layer and `GetDisplayStringForColumnValues` uses the cache in the newly defined FAR column which implements this - so it should be efficient.